lib/improver/weighted_blend.py

```diff
@@ -33,13 +33,208 @@
 
 import numpy as np
 import iris
+from iris.analysis import Aggregator
 
 
-class BasicWeightedAverage(object):
-    """Apply a Basic Weighted Average to a cube."""
+class BlendingUtilities(object):
+    """Class for blending utilities functions"""
+
+    def __init__(self):
+        """
+        Initialise class.
+        """
+        pass
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<BlendingUtilities>')
+        return result
+
+    @staticmethod
+    def basic_weighted_average(cube, weights, coord, coord_dim):
+        """Calculate weighted mean across the chosen coord
+
+        Args:
+            cube : iris.cube.Cube
+                   Cube to blend across the coord.
+            weights: Optional list or np.array of weights
+                     or None (equivalent to equal weights)
+            coord : string
+                     The name of a coordinate dimension in the cube.
+            coord_dim : tuple
+                     The index of the coordinate dimension in the cube.
+
+        Returns:
+            result : iris.cube.Cube
+                     containing the weighted mean across the chosen coord
+        """
+        # Supply weights as an array of weights whose shape matches the cube.
+        weights_array = None
```

Comment:
I think this is ok, although see my other comment on the treatment of weights input as weights